sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzMonListExtensions.cs

```diff
@@ -299,6 +313,10 @@ internal static (string? DbName, string? DbTarget) GetDbDependencyTargetAndName(
                 target = peerService;
             }
             if (string.IsNullOrWhiteSpace(target))
+            {
+                target = tagObjects.GetTargetUsingServerAttributes(defaultPort);
+            }
+            if (string.IsNullOrWhiteSpace(target))
```

Comment:
should these be `else if`?

Reply:
No, did some refactor though. Please review.